hicommonwealth/commonwealth
# Comments overhaul

## Link to Issue
Closes: #4609 #4608 #4607 #5155 

## Description of Changes
For visibility and readability each of three ticket had its own PR 
  - https://github.com/hicommonwealth/commonwealth/pull/5134 
  -  https://github.com/hicommonwealth/commonwealth/pull/5129, 
  -  https://github.com/hicommonwealth/commonwealth/pull/5109)  
 
 Now, they are all combined into single one against master 

TLDR:
- added tooltip for reply comment button
- changed icons in some places
- changed the way how the publish date is displayed
- added tooltip with formatted date of publish
 - adjusted styles of vertical lines for desktop and mobile
   - for this had to recreate how we construct comments tree. It was done recursively but in the rendering phase. I moved this step before, so now the list of comments is created first, and then it is rendered in this particular order. I was necessary because I had to know what is the next comment in the list, to be able to display vertical lines properly.  [Here is the design](https://github.com/hicommonwealth/commonwealth/pull/5138#issuecomment-1736442275) request for this piece.


## Test Plan
- go to thread page and try to add bunch of top level comments and nested comments
- you should not be able to nest more than 8 comments (reply button disabled + tooltip visible)
- some icons have been changed (share popover => link url, twitter and edit comment icon)
- address tag is not visible next to the comment's author
- time of the comment is now relative, exact time is displayed in the tooltip
- check also mobile if everything looks good
- this PR is only about UI changes, so there should not be any regression

## Deployment Plan

1. regular deployment

## Other Considerations

- needs design/product review
